### PR TITLE
Add SalesforceApexRestOperator

### PR DESCRIPTION
**Pull request by mariotaddeucci**
Add SalesforceApexRestOperator, this operator allows to call Apex rest methods

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/salesforce/hooks/salesforce.py [hunk 1]
```diff
@@ -171,9 +171,7 @@ def make_query(
         query_params = query_params or {}
         query_results = conn.query_all(query, include_deleted=include_deleted, **query_params)
 
-        self.log.info(
-            "Received results: Total size: %s; Done: %s", query_results['totalSize'], query_results['done']
-        )
+        self.log.info("Received results: Total size: {totalSize}; Done: {done}".format(query_results))
```
Comment (uranusjr) [inline airflow/providers/salesforce/hooks/salesforce.py]: The previous code is correct.

https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/

The article is about f-strings, but also applies to eager `.format()` calls.

**Review by uranusjr — CHANGES_REQUESTED** [review]
I don't unsderstand Salesforce to say anything about the operator logic, so a comment on Python code in general.

**Review by uranusjr — APPROVED** [review]
Again, can't comment on the actual operator, but the usage lgtm 🙂

**Comment by mariotaddeucci** [comment]
I'm checking the docs to pass on tests

**Review by potiuk — APPROVED** [review]
LGTM. Nice to see examples, howtos etc. ! Good job.

**Pull request merged**